homeassistant/components/tplink/coordinator.py

```diff
@@ -7,6 +7,7 @@
 
 from kasa import AuthenticationException, SmartDevice, SmartDeviceException
 
+from homeassistant import config_entries
```

Comment:
narrow the type in `TPLinkDataUpdateCoordinator` instead since it can never be `None` as the coordinator is always created from a config entry

```diff
@@ -21,6 +21,8 @@ REQUEST_REFRESH_DELAY = 0.35
 class TPLinkDataUpdateCoordinator(DataUpdateCoordinator[None]):
     """DataUpdateCoordinator to gather data for a specific TPLink device."""
 
+    config_entry: config_entries.ConfigEntry
+
     def __init__(
         self,
         hass: HomeAssistant,
@@ -40,9 +42,6 @@ class TPLinkDataUpdateCoordinator(DataUpdateCoordinator[None]):
```